app/src/main/cpp/skyline/vfs/nacp.cpp

```diff
@@ -1,21 +1,37 @@
 // SPDX-License-Identifier: MPL-2.0
 // Copyright © 2020 Skyline Team and Contributors (https://github.com/skyline-emu/)
 
+#include <bit>
 #include "nacp.h"
 
 namespace skyline::vfs {
     NACP::NACP(const std::shared_ptr<vfs::Backing> &backing) {
         nacpContents = backing->Read<NacpData>();
 
-        // TODO: Select based on language settings, complete struct, yada yada
-
-        // Iterate till we get to the first populated entry
+        u32 counter{0};
         for (auto &entry : nacpContents.titleEntries) {
-            if (entry.applicationName.front() == '\0')
-                continue;
-
-            applicationName = span(entry.applicationName).as_string(true);
-            applicationPublisher = span(entry.applicationPublisher).as_string(true);
+            if (entry.applicationName.front() != '\0') {
+                supportedTitleLanguages |= (1 << counter);
```

Comment:
This is fine?

Reply:
Nope, `counter` type could be any value big enough to reach the value `16` and it's only used as a loop index. I chose to make it `u32` but it could also be `u16` or `u8`.